neevaco/neeva-android
# Adds obfuscatedUserID timeout so login loading cannot load forever

We have no guarantee that the `InitializeGooglePlaySubscriptionMutation` which generates an `ObfuscatedUserID` won't ever hang, so we need to make it possible for this ApolloGraphQL call to timeout.

In situations where the internet connection is super slow and we never get a response back, this PR is necessary (even if the back-end has a timeout). _**Although it does make sense to just prohibit the user from logging in and skipping the Billing Flow if their internet connection is this slow... I'll follow the 1 minute timeout pattern we used for #1199 **_ 
* If the user wants to buy premium but the obfuscated user id generation timed out, I trust that they will want to buy it at another time and will when their internet is more stable

Also: 
* Adds more assume failure cases for ObfuscatedUserID
* Ran into some race conditions between the obfuscatedUserID generation and the other sign jobs when testing. The way we built the NeevaUser `startSignInJobConsumer` was that the job queue won't move forward until a job has finished. In this case:
  * There was the buy() job that was queuing a onSignedIn job (which shouldn't have happened because I kept selecting "FREE"),
    *  I've made it so that all the navigation after the sign in logic happens in the `WelcomeFlowNavModel` (see: `navigateToLoginResultScreenWhenReady()`)
    * Now `queueBuyPremiumOnSignIn` only happens when a premium subscription tag is selected.
  * Fetch Obfuscated UserID does not need to need to be a `SignInJob`
* Tests (also added timeout tests)

## Checklists

### How was this tested?

### Manual test cases
Login but manually add a delay in the InitializeGooglePlaySubscriptionMutation so it takes too long.

Login without modifying the code and see that it works fine.

### Screenshots and screen recordings
Login but manually add a delay in the InitializeGooglePlaySubscriptionMutation so it takes too long.


https://user-images.githubusercontent.com/20916043/215393699-8402e24e-4d8b-4772-b853-652cd397c119.mp4





Login without modifying the code and see that it works fine. (on debug -> loading screen will move on because billing unavailable)

https://user-images.githubusercontent.com/20916043/215339470-9caf6576-1ad8-47bd-9acc-aa02312f779d.mp4



Login without modifying the code and see that it works fine. (on release -> loading screen will move on because obfuscated userid is valid + billing is enabled)


https://user-images.githubusercontent.com/20916043/215379521-9b5ae3eb-697f-4302-8227-172c666bf1de.mp4




## Issues addressed
* Also fixes #1198
